resources/tests/org.eclipse.core.tests.resources/src/org/eclipse/core/tests/resources/CharsetTest.java

```diff
@@ -193,6 +195,45 @@ public void _testBug67606() throws CoreException {
 		}
 	}
 
+	@Test
+	public void testCharsetMoveOnFileMove() throws CoreException {
+		IWorkspace workspace = getWorkspace();
+		final IProject project = workspace.getRoot().getProject("MyProject");
+		try {
+			final IFile file = project.getFile("file.txt");
+			ensureExistsInWorkspace(file, true);
+			project.setDefaultCharset("FOO", getMonitor());
+			assertEquals("0.9", "FOO", file.getCharset());
+			file.setCharset("BAR", getMonitor());
+			assertEquals("1.0", "BAR", file.getCharset());
+			file.move(project.getFullPath().append("file2.txt"), IResource.NONE, getMonitor());
+			IFile file2 = project.getFile("file2.txt");
+			assertExistsInWorkspace(file2, false);
+			assertEquals("2.0", "BAR", file2.getCharset());
```

Comment:
Could you please add proper messages for the assertions, such as "file has not the expected default charset" for the first one?